src/boundary_conditions.jl

```diff
@@ -54,19 +120,17 @@ getbc(bc::BC{C, <:Function}, args...)            where C = bc.condition(args...)
 
 Base.getindex(bc::BC{C, <:AbstractArray}, inds...) where C = getindex(bc.condition, inds...)
 
-
-
 #####
 ##### Boundary conditions along particular coordinates
 #####
 
 """
-    CoordinateBoundaryConditions(left, right)
+    CoordinateBoundaryConditions{L, R}(left, right)
```

Comment:
While showing the types explicitly in the constructor syntax is technically correct, is this what we want? 

For *users*, it's almost certain that the type parameters `L` and `R` are always be inferred, and thus user doesn't need to know what they are. The julia documentation does allow that the function signature in the docs "can be identical to the signature present in the Julia code (like mean(x::AbstractArray)), or a simplified form. "